src/hs/Text/Regex/Pcre2/Internal.hs

```diff
@@ -1140,8 +1140,8 @@ getAllSliceRanges matchDataPtr = do
 
 -- | Helper to create non-Template Haskell API functions.  They all take options
 -- and a pattern, and then do something via a 'Matcher'.
-withMatcher :: (Matcher -> a) -> Option -> Text -> a
-withMatcher f option patt = f $ unsafePerformIO $ assembleMatcher option patt
+newMatcher :: Option -> Text -> Matcher
+newMatcher option patt = unsafePerformIO $ assembleMatcher option patt
```

Comment:
The motivation for this function was only the enabling of point-free programming, so let's delete it.